server/src-lib/Hasura/App.hs

```diff
@@ -643,57 +644,58 @@ instance HttpLog ServerAppM where
 
-instance MonadExecuteQuery ServerAppM where
+instance MonadExecuteQuery PGMetadataStorageApp where
   cacheLookup _ _ = pure ([], Nothing)
   cacheStore  _ _ = pure ()
 
-instance UserAuthentication (Tracing.TraceT ServerAppM) where
+instance UserAuthentication (Tracing.TraceT PGMetadataStorageApp) where
   resolveUserInfo logger manager headers authMode =
     runExceptT $ getUserInfoWithExpTime logger manager headers authMode
 
-instance MetadataApiAuthorization ServerAppM where
+instance MetadataApiAuthorization PGMetadataStorageApp where
   authorizeMetadataApi query userInfo = do
     let currRole = _uiRole userInfo
     when (requiresAdmin query && currRole /= adminRoleName) $
       withPathK "args" $ throw400 AccessDenied errMsg
     where
       errMsg = "restricted access : admin only"
 
-instance ConsoleRenderer ServerAppM where
+instance ConsoleRenderer PGMetadataStorageApp where
   renderConsole path authMode enableTelemetry consoleAssetsDir =
     return $ mkConsoleHTML path authMode enableTelemetry consoleAssetsDir
 
-instance MonadGQLExecutionCheck ServerAppM where
+instance MonadGQLExecutionCheck PGMetadataStorageApp where
   checkGQLExecution userInfo _ enableAL sc query = runExceptT $ do
     req <- toParsed query
     checkQueryInAllowlist enableAL userInfo req sc
     return req
 
-instance MonadConfigApiHandler ServerAppM where
+instance MonadConfigApiHandler PGMetadataStorageApp where
   runConfigApiHandler = configApiGetHandler
 
-instance MonadQueryLog ServerAppM where
+instance MonadQueryLog PGMetadataStorageApp where
   logQueryLog logger query genSqlM reqId =
     unLogger logger $ QueryLog query genSqlM reqId
 
-instance WS.MonadWSLog ServerAppM where
+instance WS.MonadWSLog PGMetadataStorageApp where
   logWSLog = unLogger
 
-runTxInMetadataStorage :: Q.TxE QErr a -> MetadataStorageT ServerAppM a
-runTxInMetadataStorage tx = do
-  pool <- lift ask
-  liftEitherM $ liftIO $ runExceptT $
-    Q.runTx pool (Q.RepeatableRead, Just Q.ReadWrite) tx
-
-instance MonadMetadataStorage (MetadataStorageT ServerAppM) where
-
-  getDeprivedCronTriggerStats        = runTxInMetadataStorage getDeprivedCronTriggerStatsTx
-  getScheduledEventsForDelivery      = runTxInMetadataStorage getScheduledEventsForDeliveryTx
-  insertScheduledEvent               = runTxInMetadataStorage . insertScheduledEventTx
-  insertScheduledEventInvocation a b = runTxInMetadataStorage $ insertInvocationTx a b
-  setScheduledEventOp a b c          = runTxInMetadataStorage $ setScheduledEventOpTx a b c
-  unlockScheduledEvents a b          = runTxInMetadataStorage $ unlockScheduledEventsTx a b
-  unlockAllLockedScheduledEvents     = runTxInMetadataStorage unlockAllLockedScheduledEventsTx
+instance MonadTx (MetadataStorageT PGMetadataStorageApp) where
+  liftTx tx = do
+    pool <- lift ask
+    -- Each operation in metadata storage is executed in an isolated transaction.
+    -- No two operations are not necessarily executed together.
```

Comment:
I think there's a negation too many in this sentence. Or I don't understand what it is saying.